src/shared_modules/content_manager/src/actionOrchestrator.hpp

```diff
@@ -142,6 +125,71 @@ class ActionOrchestrator final
         m_spBaseContext->downloadedFileHash.clear();
     }
 
+    /**
+     * @brief Creates and triggers a new orchestration that updates the offset in the database.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param offset New value of the offset.
+     */
+    void runOffsetUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, int offset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' offset update", m_spBaseContext->topicName.c_str());
+
+        if (0 > offset)
+        {
+            throw std::invalid_argument {"Invalid offset value: " + std::to_string(offset)};
+        }
+
+        spUpdaterContext->currentOffset = offset;
+
+        FactoryOffsetUpdater::create(m_spBaseContext->configData)->handleRequest(spUpdaterContext);
+    }
+
+    /**
+     * @brief Triggers a new orchestration that updates the content.
+     *
+     * @param spUpdaterContext Updater context.
+     * @param resetOffset If true, the current offset is set to zero.
+     */
+    void runContentUpdate(std::shared_ptr<UpdaterContext> spUpdaterContext, const bool resetOffset) const
+    {
+        logDebug2(WM_CONTENTUPDATER, "Running '%s' content update", m_spBaseContext->topicName.c_str());
+
+        // If the database exists, get the last offset
+        if (m_spBaseContext->spRocksDB)
+        {
+            spUpdaterContext->currentOffset = std::stoi(
+                m_spBaseContext->spRocksDB->getLastKeyValue(Components::Columns::CURRENT_OFFSET).second.ToString());
+        }
+
+        if (resetOffset)
+        {
+            spUpdaterContext->currentOffset = 0;
+        }
+
+        // If an offset download is requested and the current offset is '0', a snapshot will be downloaded with
+        // the full content to avoid downloading many offsets at once.
+        const auto& contentSource {m_spBaseContext->configData.at("contentSource").get_ref<const std::string&>()};
+        if (0 == spUpdaterContext->currentOffset && "cti-offset" == contentSource)
+        {
+            runFullContentDownload(spUpdaterContext);
+        }
+
+        // Store last file hash.
+        const auto lastDownloadedFileHash {m_spBaseContext->downloadedFileHash};
+
+        // Run the updater chain
+        m_spUpdaterOrchestration->handleRequest(spUpdaterContext);
+
+        // Update filehash if it has changed.
+        if (m_spBaseContext->spRocksDB && m_spBaseContext->downloadedFileHash != lastDownloadedFileHash)
```

Comment:
Nitpick: Align with the rest of the method (use of yoda notation)
```suggestion
        if (m_spBaseContext->spRocksDB && lastDownloadedFileHash != m_spBaseContext->downloadedFileHash)
```

Reply:
Done.